### PR TITLE
Bind reactor to thread

### PR DESCRIPTION
**Pull request by vlopes11**
**Describe the change**
This commit introduces the default bind behavior of the reactor to have its loop in a dedicated thread.

Previously, this was left for the user to decide, but then we might want to avoid complicated schemes to interact with the created service such as admin channels. Instead, the bind function will simply return the allocated socket for the listener, and any interaction with the reactor backend should be done from that entrypoint.

**Introduced tech-debts**
The user will no longer control the thread allocation of the bind service. However, there is no clear use-case of when the user might want to do that

**Base issue**
Resolves #0

**Related issues**

**Additional context**

**Checklist**
- [x] If its a bug, the branch is `bug/<user>/<issue>-short-title`. Example: `bug/vlopes11/28-stack-overflow`
- [x] If its a feature, the branch is `feature/<user>/<issue>-short-title`. Example: `feature/vlopes11/83-dynamic-menu`
- [x] I have performed a self-review of my code
- [x] If its a bug, I have added negative tests that will demonstrate it is fixed.
- [x] If its a feature, I have added unit and integrated tests.
- [x] I have succinctly documented the changes.
- [x] I have added the label `breaking change` if a public API changed either its signature or output logic.
- [x] If I used `unsafe` code, I added a comment on the line immediately above with a `Safety` remark, explaining why its usage won't cause undefined behavior if the user follow the constraints described in the documentation.
- [x] If my public function can panic, I added a `# Panics` section in the function documentation.

### CODE REVIEW

File: src/reactor.rs [hunk 1]
```diff
@@ -89,225 +90,235 @@ where
 }
 
 /// Listen in a socket using the provided capacity for the used channels
-async fn bind<B, S>(capacity: usize, socket: S) -> io::Result<()>
+async fn bind<B, S>(capacity: usize, socket: S) -> io::Result<SocketAddr>
```
Comment (Daksh14) [inline src/reactor.rs:93]: why does this return `SocketAddr`? Won't this function exit when the server is stopped?

File: src/reactor.rs [hunk 2]
```diff
@@ -89,225 +90,235 @@ where
...
+                            {
+                                tracing::error!("error sending reverse request: {}", e);
+                            }
                         }
-                    }
-                });
-
-                // thread to handle inbound requests to be processed by the backend
-                tokio::spawn(async move {
-                    let mut backend = B::init(events_tx, requests_tx).await;
```
Comment (Daksh14) [inline src/reactor.rs:177]: **nitpitk** I would recommend you import Backend like `use crate::Backend as BackendTrait` and rename `B` to `Backend`, this makes it more readable. 
```rust
let mut backend = Backend::init(events_tx, requests_tx).await;
```

File: src/reactor.rs [hunk 3]
```diff
@@ -89,225 +90,235 @@ where
...
 
-            Err(e) => tracing::error!("error accepting socket: {}", e),
+                Err(e) => tracing::error!("error accepting socket: {}", e),
+            }
         }
-    }
+    });
+
+    Ok(addr)
 }
```
Comment (Daksh14) [inline src/reactor.rs:324]: I haven't caught any mistakes in the protocol implementation yet but we will encounter them along the way and also when the extension is created from the client side. Let's push this implementation now after we're done with the other changes :)

**Review by Daksh14 — CHANGES_REQUESTED** [review]

**Comment by vlopes11** [comment]
Superseded by #10

**Pull request closed without merging**